formver.compiler-plugin/uniqueness/src/org/jetbrains/kotlin/formver/uniqueness/UniqueCheckVisitor.kt

```diff
@@ -111,10 +120,10 @@ object UniqueCheckVisitor : FirVisitor<Pair<UniqueLevel, UniquePathContext?>, Un
                         "attempting to pass a partially shared argument ${argument.source.text} in ${functionCall.source.text}"
                     }
 
-                    trie?.level = UniqueLevel.Top
+                    if (paramBorrowing != BorrowingLevel.Borrowed) trie?.level = UniqueLevel.Top
                 }
 
-                UniqueLevel.Shared -> trie?.level = UniqueLevel.Shared
+                UniqueLevel.Shared -> if (paramBorrowing != BorrowingLevel.Borrowed) trie?.level = UniqueLevel.Shared
```

Comment:
What if it is borrowed?  Is that caught earlier?  I think a comment may be reasonable